### DISCUSSION TITLE
Have `createSystem` extend the default system values

### DISCUSSION THREAD

**Discussion by scottrepreneur** [wevm/frog-archived#328] [Ideas]
### Describe the bug

I am looking to extend the existing system values, so I only need to add my own custom ones in specific cases.

```ts
import { createSystem } from "frog/ui";

export const {
  Heading,
  vars,
  ...
} = createSystem({
  colors: {
    gradientBg: "linear-gradient(to right, #432889, #17101F)",
  }
});
```

This leaves only one color in the system it seems, instead of extending the default color set with this additional color.

<img width="833" alt="Screenshot 2024-05-19 at 10 09 51" src="https://github.com/wevm/frog/assets/1778380/c65ec2ac-39e9-433e-9d26-4f7326745154">

### Steps To Reproduce

Lmk if you need this repro. Uses the default UI setup with an additional color as described above.

### Frog Version

0.11.2

### TypeScript Version

5.4.5

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/frog/issues) for the bug I encountered.

**Comment by tmm** [comment]
Check out `defaultVars`. In the future, we could add an [`extend` property a la Tailwind](https://tailwindcss.com/docs/theme#extending-the-default-theme).

```ts
import { createSystem, defaultVars } from 'frog/ui'

export const { Heading, vars } = createSystem({
  colors: {
    ...defaultVars.colors,
    gradientBg: 'linear-gradient(to right, #432889, #17101F)',
  },
})
```

**Reply by scottrepreneur** [reply]
Could add `defaultVars` mention to the docs 🙏  Colors is only mentioned under [theme mode](https://frog.fm/ui/ui-system#light--dark-mode)

**Reply by tmm** [reply]
Adding to my list. PR welcome in the meantime!
